app/services/SubmissionService.scala

```diff
@@ -66,11 +67,11 @@ class SubmissionService @Inject()(
 
   private def sendMovementRequest(
     choice: Choice,
-    data: InventoryLinkingMovementRequest
+    data: MovementRequest
   )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] =
     choice match {
-      case Arrival   => connector.sendArrivalDeclaration(data.toXml)
-      case Departure => connector.sendDepartureDeclaration(data.toXml)
+      case Arrival   => connector.sendArrivalDeclaration(Json.toJson(data).toString())
```

Comment:
Json.toString is asking for trouble

Reply:
Agreed, can we pass the model to the connector please

Reply:
Done